src/waveform/renderers/allshader/waveformrendererrgb.cpp

```diff
@@ -140,21 +145,29 @@ void WaveformRendererRGB::paintGL() {
         float maxMid = static_cast<float>(u8maxMid);
         float maxHigh = static_cast<float>(u8maxHigh);
         float maxAllChn[2]{static_cast<float>(u8maxAllChn[0]), static_cast<float>(u8maxAllChn[1])};
+
+        float ghostLow = maxLow;
+        float ghostMid = maxMid;
+        float ghostHigh = maxHigh;
+        float maxAllChn_ghost[2]{static_cast<float>(u8maxAllChn[0]),
+                static_cast<float>(u8maxAllChn[1])};
         // Uncomment to undo scaling with pow(value, 2.0f * 0.316f) done in analyzerwaveform.h
         // float maxAllChn[2]{unscale(u8maxAllChn[0]), unscale(u8maxAllChn[1])};
 
         // Calculate the squared magnitude of the maxLow, maxMid and maxHigh values.
         // We take the square root to get the magnitude below.
         const float sum = math_pow2(maxLow) + math_pow2(maxMid) + math_pow2(maxHigh);
 
-        // Apply the gains
+        // Apply the gains (non-ghosted version only)
         maxLow *= lowGain;
         maxMid *= midGain;
         maxHigh *= highGain;
 
         // Calculate the squared magnitude of the gained maxLow, maxMid and maxHigh values
         // We take the square root to get the magnitude below.
         const float sumGained = math_pow2(maxLow) + math_pow2(maxMid) + math_pow2(maxHigh);
+        const float sumGained_ghost = math_pow2(ghostLow) +
```

Comment:
You are overcomplicating things. You don't need to sum the low+mid+high components because ... (see below)